axis_backend/apps/services_app/services/service_assignment_service.py

```
@@ -0,0 +1,139 @@
+        frequency: str,
+        end_date: Optional[date] = None,
+        status: str = AssignmentStatus.PENDING,
+        metadata: Optional[dict] = None,
+        **kwargs
+    ) -> ServiceAssignment:
+        """Create a new service assignment."""
+        # Validate service exists
+        try:
+            service = Service.objects.get(id=service_id)
+        except Service.DoesNotExist:
+            raise ValidationError(f"Service with ID '{service_id}' does not exist")
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)